### PR TITLE
broaden wartremover application

### PR DESCRIPTION
**Pull request by drostron**

### CODE REVIEW

File: web/src/main/scala/quasar/api/server.scala [hunk 1]
```diff
@@ -159,6 +157,7 @@ object Server {
     openClient: Boolean,
     port: Option[Int])
 
+  @SuppressWarnings(Array("org.brianmckenna.wartremover.warts.NonUnitStatements"))
```
Comment (sellout) [inline web/src/main/scala/quasar/api/server.scala]: Is this also due to puffnfresh/wartremover#149? We should have a comment here explaining why we need the suppression.

File: web/src/main/scala/quasar/api/zip.scala [hunk 2]
```diff
@@ -54,6 +54,7 @@ object Zip {
     // mutating operations: one to accept an Op to be processed, and another
     // to poll for data that's been written.
     class Buffer {
+      @SuppressWarnings(Array("org.brianmckenna.wartremover.warts.Var"))
```
Comment (sellout) [inline web/src/main/scala/quasar/api/zip.scala]: And a comment here explaining why we need a var.

Suppressions should come with explanations, basically.

File: web/src/main/scala/quasar/api/server.scala (hunk 1 above)
Comment (sellout) [inline web/src/main/scala/quasar/api/server.scala]: Oh, no … not 149, but still.

**Comment by mossprescott** [comment]
This is a good idea. Any idea if it has any impact on compile time, or on how often we get a total recompile?

**Comment by drostron** [comment]
I'll add comments as suggested by @sellout.

@mossprescott, the build times appear to be within the standard range and the incremental compilation situation appears to be similar. _(tough to be precise on those at the moment though)_

**Comment by drostron** [comment]
Speaking of the incremental compilation situation, I spent a bit of time on it and appear to have found a solution. The reason it works eludes me unfortunately. Hopefully it address the issue for others as well.

**Comment by sellout** [comment]
I’m ok with that as a workaround, but we should open an issue against wartremover for it, too, so we can go back to the other way eventually.

**Comment by drostron** [comment]
Agreed on opening an issue against wartremover. I'll look into getting a small repo case setup and submit an issue.

**Comment by drostron** [comment]
incremental compilation wartremover issue: puffnfresh/wartremover/issues/202

**Comment by drostron** [comment]
Pushed commit adding comments for bare @SuppressWarnings.

**Comment by mossprescott** [comment]
Since your PR was merged already, do we want to wait for a wart-remover release as opposed to merging this workaround? Otherwise, we'll need to remember to come back and clean this up, right?

**Comment by drostron** [comment]
It could be a month or two before the next wartremover release and it might be nice to have incremental compilation working before then. We will have to remember to come back to it, but I imagine the associated comment will remind whomever makes the next wartremover settings update.

**Pull request merged**